File: agent/app/dto/firewall.go
package dto

import (
	"github.com/1Panel-dev/1Panel/agent/utils/firewall/filter"
	firewallsync "github.com/1Panel-dev/1Panel/agent/utils/firewall/sync"
)

type FirewallSubsystemStatus struct {
	Name            string                      `json:"name"`
	Backend         string                      `json:"backend"`
	ConflictBackend string                      `json:"conflictBackend,omitempty"`
	IsExist         bool                        `json:"isExist"`
	IsActive        bool                        `json:"isActive"`
	IsInit          bool                        `json:"isInit"`
	IsBind          bool                        `json:"isBind"`
	Version         string                      `json:"version"`
	PingStatus      string                      `json:"pingStatus"`
	Message         string                      `json:"message,omitempty"`
	Reason          string                      `json:"reason,omitempty"`
	SyncError       string                      `json:"syncError,omitempty"`
	IPv4            FirewallBackendFamilyStatus `json:"ipv4"`
	IPv6            FirewallBackendFamilyStatus `json:"ipv6"`
}

type FirewallLifecycleOperation struct {
	Operation         string `json:"operation" validate:"required,oneof=start stop restart disableBanPing enableBanPing"`
	WithDockerRestart bool   `json:"withDockerRestart"`
}

type FirewallBackendOption struct {
	Name           string                      `json:"name"`
	Installed      bool                        `json:"installed"`
	Active         bool                        `json:"active"`
	Initialized    bool                        `json:"initialized"`
	Bound          bool                        `json:"bound"`
	Supported      bool                        `json:"supported"`
	SupportReason  string                      `json:"supportReason,omitempty"`
	Implementation string                      `json:"implementation,omitempty"`
	Message        string                      `json:"message,omitempty"`
	IPv4           FirewallBackendFamilyStatus `json:"ipv4"`
	IPv6           FirewallBackendFamilyStatus `json:"ipv6"`
}

type FirewallBackendFamilyStatus struct {
	Available   bool   `json:"available"`
	Initialized bool   `json:"initialized"`
	Bound       bool   `json:"bound"`
	Reason      string `json:"reason,omitempty"`
}

type FirewallBackendGroup struct {
	Selected string                  `json:"selected"`
	Current  string                  `json:"current,omitempty"`
	Options  []FirewallBackendOption `json:"options"`
}

type FirewallSettings struct {
	System        FirewallBackendGroup `json:"system"`
	Forwarding    FirewallBackendGroup `json:"forwarding"`
	Docker        FirewallBackendGroup `json:"docker"`
	PingStatus    string               `json:"pingStatus"`
	PortWhitelist string               `json:"portWhiteList"`
}

type FirewallBackendOperation struct {
	Subsystem string `json:"subsystem" validate:"required,oneof=system forwarding docker"`
	Backend   string `json:"backend" validate:"required,oneof=firewalld ufw iptables nftables"`
	Operation string `json:"operation" validate:"required,oneof=select initialize cleanup"`
}

type FilterChainOperation struct {
	Name    string `json:"name" validate:"required,eq=1PANEL_BASIC"`
	Operate string `json:"operate" validate:"required,oneof=init-base bind-base unbind-base"`
}

type FirewallSystemPort struct {
	Family   string
	Port     string
	Protocol string
}

type FirewallRuleInventoryResponse struct {
	Total        int64                  `json:"total"`
	AllTotal     int64                  `json:"allTotal"`
	ManagedTotal int64                  `json:"managedTotal"`
	Items        []filter.InventoryItem `json:"items"`
	Notices      []filter.ScopeNotice   `json:"notices,omitempty"`
}

type FirewallRuleResetResponse struct {
	Removed  int  `json:"removed"`
	Disabled bool `json:"disabled"`
}

type FirewallRuleReset struct {
	Provider filter.Provider `json:"provider,omitempty" validate:"omitempty,oneof=firewalld ufw iptables nftables"`
}

type FirewallRuleCheckResult struct {
	Decision         filter.CheckDecision       `json:"decision"`
	Classification   filter.CheckClassification `json:"classification"`
	Reason           string                     `json:"reason"`
	RequestedRule    filter.FirewallRule        `json:"requestedRule"`
	RequestedRuleKey string                     `json:"requestedRuleKey"`
	ExistingRuleUUID string                     `json:"existingRuleUUID,omitempty"`
	Candidates       []filter.ObservedRule      `json:"candidates,omitempty"`
	AllowedActions   []filter.CheckAction       `json:"allowedActions,omitempty"`
	CheckFlag        string                     `json:"checkFlag"`
}

type FirewallRuleInventory struct {
	PageInfo
	Scope         filter.Scope            `json:"scope,omitempty"`
	Scopes        []filter.Scope          `json:"scopes,omitempty" validate:"max=16"`
	All           bool                    `json:"all,omitempty"`
	Info          string                  `json:"info"`
	Families      []filter.Family         `json:"families,omitempty" validate:"omitempty,dive,oneof=ipv4 ipv6"`
	Actions       []string                `json:"actions,omitempty" validate:"omitempty,dive,oneof=accept deny"`
	States        []filter.InventoryState `json:"states,omitempty" validate:"omitempty,dive,oneof=managed adopted external drifted protected"`
	ExcludeChains []string                `json:"excludeChains,omitempty" validate:"omitempty,dive,oneof=1PANEL_BASIC_BEFORE 1PANEL_BASIC 1PANEL_BASIC_AFTER"`
}

type FirewallNativeDetail struct {
	Provider   filter.Provider   `json:"provider" validate:"required,oneof=firewalld ufw"`
	NativeKind filter.NativeKind `json:"nativeKind" validate:"required,oneof=zone_service ufw_application"`
	Name       string            `json:"name" validate:"required"`
	Permanent  bool              `json:"permanent"`
}

type DockerPortGuardBase struct {
	Name        string                      `json:"name"`
	Version     string                      `json:"version"`
	IsExist     bool                        `json:"isExist"`
	Initialized bool                        `json:"initialized"`
	Bound       bool                        `json:"bound"`
	IPv4        DockerPortGuardFamilyStatus `json:"ipv4"`
	IPv6        DockerPortGuardFamilyStatus `json:"ipv6"`
	Backend     string                      `json:"backend"`
	Message     string                      `json:"message,omitempty"`
}

type DockerPortGuardFamilyStatus struct {
	State       string `json:"state"`
	Reason      string `json:"reason,omitempty"`
	Initialized bool   `json:"initialized"`
	Bound       bool   `json:"bound"`
	Effective   bool   `json:"effective"`
}

type DockerPortGuardEndpoint struct {
	Family        string   `json:"family"`
	HostIP        string   `json:"hostIP"`
	HostPort      uint16   `json:"hostPort"`
	Protocol      string   `json:"protocol"`
	ContainerID   string   `json:"containerID"`
	ContainerName string   `json:"containerName"`
	ContainerPort uint16   `json:"containerPort"`
	Compose       string   `json:"compose,omitempty"`
	Application   string   `json:"application,omitempty"`
	PolicyUUID    string   `json:"policyUUID,omitempty"`
	Mode          string   `json:"mode,omitempty"`
	Sources       []string `json:"sources"`
	Effective     bool     `json:"effective"`
	Description   string   `json:"description,omitempty"`
}

type DockerPortGuardPortGroup struct {
	Key       string                    `json:"key"`
	Label     string                    `json:"label"`
	Endpoint  DockerPortGuardEndpoint   `json:"endpoint"`
	Endpoints []DockerPortGuardEndpoint `json:"endpoints"`
}

type DockerPortGuardContainer struct {
	Key         string                     `json:"key"`
	Name        string                     `json:"name"`
	Compose     string                     `json:"compose,omitempty"`
	Application string                     `json:"application,omitempty"`
	Endpoints   []DockerPortGuardEndpoint  `json:"endpoints"`
	PortGroups  []DockerPortGuardPortGroup `json:"portGroups"`
}

type DockerPortGuardList struct {
	Base           DockerPortGuardBase        `json:"base"`
	Containers     []DockerPortGuardContainer `json:"containers"`
	OrphanPolicies []DockerPortGuardEndpoint  `json:"orphanPolicies"`
}

type DockerPortGuardEndpointIdentity struct {
	Family   string `json:"family" validate:"required,oneof=ipv4 ipv6"`
	HostIP   string `json:"hostIP" validate:"required,max=45"`
	HostPort uint16 `json:"hostPort" validate:"required,min=1"`
	Protocol string `json:"protocol" validate:"required,oneof=tcp udp"`
}

type DockerPortGuardPolicyBatch struct {
	Endpoints   []DockerPortGuardEndpointIdentity `json:"endpoints" validate:"required,min=1,max=256,dive"`
	Mode        string                            `json:"mode" validate:"required,oneof=deny_sources allow_sources deny_all"`
	Sources     []string                          `json:"sources" validate:"max=256,dive,required,max=64"`
	Description string                            `json:"description" validate:"max=256"`
}

type DockerPortGuardPolicyBatchDelete struct {
	UUIDs []string `json:"uuids" validate:"required,min=1,max=256,dive,required,max=64"`
}

type DockerPortGuardOperation struct {
	Operation string `json:"operation" validate:"required,oneof=initialize bind unbind"`
}

type FirewallRuleCheckItem struct {
	UUID string              `json:"uuid" validate:"omitempty,max=64"`
	Rule filter.FirewallRule `json:"rule" validate:"required"`
}

type FirewallRuleCheck struct {
	Items []FirewallRuleCheckItem `json:"items" validate:"required,min=1,max=256,dive"`
}

type FirewallRuleCheckResponse struct {
	Items []FirewallRuleCheckResult `json:"items"`
}

type FirewallRuleCreateItem struct {
	Rule             filter.FirewallRule `json:"rule" validate:"required"`
	CheckFlag        string              `json:"checkFlag"`
	Action           filter.CheckAction  `json:"action"`
	AdoptInstanceKey string              `json:"adoptInstanceKey"`
	SourceKind       string              `json:"sourceKind" validate:"omitempty,oneof=user imported"`
	SourceID         string              `json:"sourceID"`
}

type FirewallRuleCreate struct {
	Items []FirewallRuleCreateItem `json:"items" validate:"required,min=1,max=256,dive"`
}

type FirewallRuleCreateResponse struct {
	Succeeded int                         `json:"succeeded"`
	Failed    int                         `json:"failed"`
	Skipped   int                         `json:"skipped"`
	Errors    []FirewallRuleCreateFailure `json:"errors,omitempty"`
}

type FirewallRuleCreateFailure struct {
	Index  int                 `json:"index"`
	Status string              `json:"status"`
	Rule   filter.FirewallRule `json:"rule"`
	Error  string              `json:"error,omitempty"`
}

type FirewallRuleSyncRequest struct {
	Subsystem      string          `json:"subsystem" validate:"omitempty,oneof=system forwarding docker"`
	SourceProvider filter.Provider `json:"sourceProvider,omitempty" validate:"omitempty,oneof=firewalld ufw iptables nftables"`
	TargetProvider filter.Provider `json:"targetProvider" validate:"required,oneof=firewalld ufw iptables nftables"`
	ResetSource    bool            `json:"resetSource"`
	TaskID         string          `json:"taskID,omitempty" validate:"omitempty,max=64"`
}

type FirewallRuleSyncItem struct {
	SourceUUID  string                   `json:"sourceUUID"`
	Rule        *filter.FirewallRule     `json:"rule,omitempty"`
	ForwardRule *ForwardRule             `json:"forwardRule,omitempty"`
	DockerRule  *DockerPortGuardEndpoint `json:"dockerRule,omitempty"`
	Status      firewallsync.Status      `json:"status"`
	ReasonCode  firewallsync.ReasonCode  `json:"reasonCode,omitempty"`
	Reason      string                   `json:"reason,omitempty"`
}

type FirewallRuleSyncPreview struct {
	Subsystem      string                 `json:"subsystem"`
	SourceProvider filter.Provider        `json:"sourceProvider,omitempty"`
	TargetProvider filter.Provider        `json:"targetProvider"`
	Total          int                    `json:"total"`
	Ready          int                    `json:"ready"`
	Existing       int                    `json:"existing"`
	Removed        int                    `json:"removed"`
	Blocked        int                    `json:"blocked"`
	Items          []FirewallRuleSyncItem `json:"items"`
}

type FirewallRuleSyncResult struct {
	Subsystem      string                    `json:"subsystem"`
	SourceProvider filter.Provider           `json:"sourceProvider,omitempty"`
	TargetProvider filter.Provider           `json:"targetProvider"`
	Total          int                       `json:"total"`
	Succeeded      int                       `json:"succeeded"`
	Skipped        int                       `json:"skipped"`
	Removed        int                       `json:"removed"`
	Failed         int                       `json:"failed"`
	Errors         []FirewallRuleSyncFailure `json:"errors,omitempty"`
	TaskID         string                    `json:"taskID,omitempty"`
	Queued         bool                      `json:"queued,omitempty"`
}

type FirewallRuleSyncTask struct {
	TaskID    string `json:"taskID,omitempty"`
	Executing bool   `json:"executing"`
}

type FirewallRuleSyncFailure struct {
	SourceUUID  string                   `json:"sourceUUID"`
	Rule        *filter.FirewallRule     `json:"rule,omitempty"`
	ForwardRule *ForwardRule             `json:"forwardRule,omitempty"`
	DockerRule  *DockerPortGuardEndpoint `json:"dockerRule,omitempty"`
	Error       string                   `json:"error"`
}

type FirewallRuleDelete struct {
	UUIDs []string `json:"uuids" validate:"required,min=1,max=256,dive,required,max=64"`
}

type FirewallRuleDeleteResponse struct {
	Succeeded int                         `json:"succeeded"`
	Failed    int                         `json:"failed"`
	Errors    []FirewallRuleDeleteFailure `json:"errors,omitempty"`
}

type FirewallRuleDeleteFailure struct {
	Index int    `json:"index"`
	UUID  string `json:"uuid"`
	Error string `json:"error"`
}

type FirewallRuleUpdate struct {
	UUID string              `json:"uuid" validate:"required,max=64"`
	Rule filter.FirewallRule `json:"rule" validate:"required"`
}

type FirewallRuleReorder struct {
	UUID           string `json:"uuid" validate:"required,max=64"`
	TargetPosition *int64 `json:"targetPosition"`
	Priority       *int   `json:"priority"`
}

File: frontend/src/api/interface/firewall.ts
import { ReqPage } from '.';

export namespace Firewall {
    export type Provider = 'iptables' | 'nftables' | 'firewalld' | 'ufw';
    export type BackendSubsystem = 'system' | 'forwarding' | 'docker';
    export type BackendOperation = 'select' | 'initialize' | 'cleanup';
    export interface BackendOption {
        name: Provider;
        installed: boolean;
        active: boolean;
        initialized: boolean;
        bound: boolean;
        supported: boolean;
        implementation?: string;
        message?: string;
        ipv4: BackendFamilyStatus;
        ipv6: BackendFamilyStatus;
    }
    export interface BackendFamilyStatus {
        available: boolean;
        initialized: boolean;
        bound: boolean;
        reason?: string;
    }
    export interface BackendGroup {
        selected: string;
        current?: string;
        options: BackendOption[];
    }
    export interface Settings {
        system: BackendGroup;
        forwarding: BackendGroup;
        docker: BackendGroup;
        pingStatus: string;
        portWhiteList: string;
    }
    export interface BackendOperateRequest {
        subsystem: BackendSubsystem;
        backend: Provider;
        operation: BackendOperation;
    }
    export interface FirewallBase {
        name: string;
        backend: string;
        conflictBackend?: string;
        isExist: boolean;
        isActive: boolean;
        isInit: boolean;
        isBind: boolean;
        version: string;
        pingStatus: string;
        syncError?: string;
        ipv4: BackendFamilyStatus;
        ipv6: BackendFamilyStatus;
    }
    export interface ForwardRuleSearch extends ReqPage {
        strategy: string;
        info: string;
    }
    export interface RuleInfo extends ReqPage {
        family: string;
        address: string;
        destination: string;
        port: string;
        srcPort: string;
        destPort: string;
        protocol: string;
        strategy: string;
        usedStatus: string;
        description: string;
        [key: string]: any;
    }
    export interface RuleForward {
        operation: string;
        family: 'ipv4' | 'ipv6';
        protocol: string;
        port: string;
        targetIP: string;
        targetPort: string;
        interface: string;
        isDesired?: boolean;
        isRuntime?: boolean;
        syncStatus?: 'converged' | 'missing' | 'runtime_only';
    }
    export type Family = 'ipv4' | 'ipv6' | 'inet';
    export type Direction = 'input';
    export type Action = 'accept' | 'drop' | 'reject';
    export type NativeKind =
        'rule' | 'zone_port' | 'rich_rule' | 'ufw_rule' | 'ufw_application' | 'opaque' | 'zone_service';
    export type ParseStatus = 'supported' | 'partial' | 'opaque';
    export type PersistenceStatus = 'converged' | 'runtime_only' | 'permanent_only';

    export interface Scope {
        provider: Provider;
        family: Family;
        table?: string;
        zone?: string;
        chain?: string;
        direction: Direction;
    }

    export interface Rule {
        uuid?: string;
        scope: Scope;
        nativeKind?: NativeKind;
        protocol: string;
        sourceAddress?: string;
        sourcePort?: string;
        destinationAddress?: string;
        destinationPort?: string;
        interface?: string;
        connectionStates?: string[];
        action: Action;
        priority?: number;
        orderIndex?: number;
        orderBucket?: string;
        description?: string;
    }

    export interface Locator {
        provider: Provider;
        scopeKey: string;
        nativeId?: string;
        canonical?: string;
        position?: number;
    }

    export interface ObservedRule {
        rule: Rule;
        locator: Locator;
        instanceKey?: string;
        marker?: string;
        parseStatus: ParseStatus;
        raw?: string;
        protected: boolean;
        persistence?: PersistenceStatus;
    }

    export type RuleOrigin = 'created' | 'adopted';
    export type InventoryState = 'managed' | 'adopted' | 'external' | 'drifted' | 'protected';
    export type InventoryMatch = 'none' | 'exact' | 'changed' | 'missing' | 'ambiguous' | 'opaque';

    export interface DesiredRule {
        uuid: string;
        rule: Rule;
        ruleKey: string;
        origin: RuleOrigin;
        marker?: string;
        observedInstanceKey?: string;
    }

    export interface RuntimeUsage {
        used: boolean;
        usedBy?: string[];
        reason?: string;
    }

    export interface InventoryItem {
        rule: Rule;
        observed?: ObservedRule;
        desired?: DesiredRule;
        state: InventoryState;
        match: InventoryMatch;
        usage?: RuntimeUsage;
    }

    export type ScopeNoticeCode =
        | 'default_scope_mismatch'
        | 'managed_scope_inactive'
        | 'unmanaged_active_scopes'
        | 'runtime_permanent_mismatch'
        | 'default_policy'
        | 'managed_scope_missing';

    export interface ScopeNotice {
        code: ScopeNoticeCode;
        values?: string[];
    }

    export interface Inventory {
        items: InventoryItem[];
        notices?: ScopeNotice[];
    }

    export interface ResetResponse {
        removed: number;
        disabled: boolean;
    }

    export interface ResetRequest {
        provider?: Provider;
    }

    export type CheckDecision = 'ready' | 'confirmation_required' | 'blocked' | 'no_change';
    export type CheckClassification =
        'none' | 'exact_managed' | 'exact_external' | 'covered' | 'conflict' | 'unsupported' | 'protected';
    export type CheckAction = 'create' | 'create_anyway' | 'adopt' | 'select_adopt' | 'cancel';
    export type ApplicableCheckAction = Exclude<CheckAction, 'cancel'>;

    export interface RuleCheckResult {
        decision: CheckDecision;
        classification: CheckClassification;
        reason: string;
        requestedRule: Rule;
        requestedRuleKey: string;
        existingRuleUUID?: string;
        candidates?: ObservedRule[];
        allowedActions?: CheckAction[];
        checkFlag: string;
    }

    export interface InventoryRequest {
        scope: Scope;
    }

    export interface NativeDetailRequest {
        provider: 'firewalld' | 'ufw';
        nativeKind: 'zone_service' | 'ufw_application';
        name: string;
        permanent: boolean;
    }

    export interface CheckItem {
        uuid?: string;
        rule: Rule;
    }

    export interface CheckRequest {
        items: CheckItem[];
    }

    export interface CheckResponse {
        items: RuleCheckResult[];
    }

    export interface CreateItem {
        rule: Rule;
        checkFlag: string;
        action: ApplicableCheckAction;
        adoptInstanceKey?: string;
        sourceKind?: 'user' | 'panel' | 'security' | 'imported';
        sourceID?: string;
    }

    export interface CreateRequest {
        items: CreateItem[];
    }

    export interface CreateResponse {
        succeeded: number;
        failed: number;
        skipped: number;
        errors?: CreateFailure[];
    }

    export interface CreateFailure {
        index: number;
        status: 'failed' | 'skipped';
        rule: Rule;
        error?: string;
    }

    export interface RuleSyncRequest {
        subsystem: BackendSubsystem;
        sourceProvider?: Provider;
        targetProvider: Provider;
        resetSource?: boolean;
        taskID?: string;
    }

    export type RuleSyncStatus = 'ready' | 'existing' | 'remove' | 'blocked';

    export interface RuleSyncItem {
        sourceUUID: string;
        rule?: Rule;
        forwardRule?: RuleForward;
        dockerRule?: DockerGuardEndpoint;
        status: RuleSyncStatus;
        reason?: string;
    }

    export interface RuleSyncPreview {
        subsystem: BackendSubsystem;
        sourceProvider?: Provider;
        targetProvider: Provider;
        total: number;
        ready: number;
        existing: number;
        removed: number;
        blocked: number;
        items: RuleSyncItem[];
    }

    export interface RuleSyncResult {
        subsystem: BackendSubsystem;
        sourceProvider?: Provider;
        targetProvider: Provider;
        total: number;
        succeeded: number;
        skipped: number;
        removed: number;
        failed: number;
        errors?: RuleSyncFailure[];
        taskID?: string;
        queued?: boolean;
    }

    export interface RuleSyncTask {
        taskID?: string;
        executing: boolean;
    }

    export interface RuleSyncFailure {
        sourceUUID: string;
        rule?: Rule;
        forwardRule?: RuleForward;
        dockerRule?: DockerGuardEndpoint;
        error: string;
    }

    export interface DeleteRequest {
        uuids: string[];
    }

    export interface DeleteResponse {
        succeeded: number;
        failed: number;
        errors?: DeleteFailure[];
    }

    export interface DeleteFailure {
        index: number;
        uuid: string;
        error: string;
    }

    export interface UpdateRequest {
        rule: Rule;
    }

    export interface ReorderRequest {
        targetPosition?: number;
        priority?: number;
    }

    export interface DockerGuardBase {
        name: string;
        version: string;
        initialized: boolean;
        bound: boolean;
        ipv4: DockerGuardFamilyStatus;
        ipv6: DockerGuardFamilyStatus;
        backend: string;
        message?: string;
    }
    export interface DockerGuardFamilyStatus {
        state: 'effective' | 'disabled' | 'not_effective';
        reason?:
            | 'command_missing'
            | 'docker_chain_missing'
            | 'guard_chain_missing'
            | 'jump_missing'
            | 'jump_not_first'
            | 'jump_duplicate'
            | 'inspect_failed';
        initialized: boolean;
        bound: boolean;
        effective: boolean;
    }
    export interface DockerGuardEndpoint {
        family: 'ipv4' | 'ipv6';
        hostIP: string;
        hostPort: number;
        protocol: 'tcp' | 'udp';
        containerID?: string;
        containerName?: string;
        containerPort?: number;
        compose?: string;
        application?: string;
        policyUUID?: string;
        mode?: 'deny_sources' | 'allow_sources' | 'deny_all';
        sources: string[];
        effective: boolean;
        description?: string;
    }
    export interface DockerGuardPortGroup {
        key: string;
        label: string;
        endpoint: DockerGuardEndpoint;
        endpoints: DockerGuardEndpoint[];
    }
    export interface DockerGuardContainer {
        key: string;
        name: string;
        compose?: string;
        application?: string;
        endpoints: DockerGuardEndpoint[];
        portGroups: DockerGuardPortGroup[];
    }
    export interface DockerGuardList {
        base: DockerGuardBase;
        containers: DockerGuardContainer[];
        orphanPolicies: DockerGuardEndpoint[];
    }
    export interface DockerGuardEndpointIdentity {
        family: 'ipv4' | 'ipv6';
        hostIP: string;
        hostPort: number;
        protocol: 'tcp' | 'udp';
    }
    export interface DockerGuardPolicy extends DockerGuardEndpointIdentity {
        mode: 'deny_sources' | 'allow_sources' | 'deny_all';
        sources: string[];
        description: string;
    }
    export interface DockerGuardPolicyBatch {
        endpoints: DockerGuardEndpointIdentity[];
        mode: 'deny_sources' | 'allow_sources' | 'deny_all';
        sources: string[];
        description: string;
    }
    export interface DockerGuardPolicyBatchDelete {
        uuids: string[];
    }
}
